conftest.py

```diff
@@ -0,0 +1,43 @@
+"""
+Configuration file for pytest
+
+See also https://docs.pytest.org/en/6.2.x/writing_plugins.html#writing-hook-functions
+"""
+
+import os
+import sys
+
+
+def pytest_sessionstart():
+    """
+    Only modify the path when run via pytest (not in the Talon runtime)
+    """
+
+    # Make the talon. stubs available to tests
+    curr_dir = os.path.dirname(__file__)
+    sys.path.append(os.path.join(curr_dir, "tests", "stubs"))
+
+    # Makes the code/ module available as knausj_code/ for use in tests. pytest uses
+    # the builtin python code module which prevents us from importing the knausj one.
+    # I've elected not to rename code as this would cause merge errors for forks that
+    # have modified it.
+    # This implementation just creates a simple symlink. An alternative might be to
+    # look at customising the python module loader:
+    # https://docs.python.org/3/reference/import.html
+    # I've had a go at relative imports without success.
+    symlink_src = os.path.join(curr_dir, "code")
```

Comment:
From aegis:

```
just need a PathFinder in sys.meta_path I think
New
8:39
looks something like this:

class PluginFinder(PathFinder):
    @classmethod
    def find_spec(cls, fullname, path=None, target=None):
        return super().find_spec(fullname, path, target)
sys.meta_path.insert(0, PluginFinder)

(edited)
8:40
if you're passed a None path, you need to use the fullname to figure out the file path. otherwise, check if the path makes sense for a path you are willing to load from (edited) 
8:40
and find_spec can return None if you couldn't find anything
perhaps:

- from knausj_code import formatters
+ from user.knausj_talon.code import formatters

8:42
so then you take any fullname starting with user.knausj_talon. and rebase it to the root of your repo
```